server/broadlistening/pipeline/steps/hierarchical_aggregation.py

```
@@ -175,21 +271,37 @@ def _build_cluster_value(melted_labels: pd.DataFrame, total_num: int) -> list[Cl
             label="全体",
             takeaway="",
-            value=total_num,
+            value=int(total_num),  # Convert to native int
             parent="",
             density_rank_percentile=0,
         )
     ]
 
     for _, melted_label in melted_labels.iterrows():
+        # Convert potential NumPy types to native Python types
+        level = int(melted_label["level"]) if isinstance(melted_label["level"], (int, np.integer)) else melted_label["level"]
```

Ruff (0.11.9)

Use `X | Y` in `isinstance` call instead of `(X, Y)`

Convert to `X | Y`

(UP038)